app/src/main/java/com/rektstudios/trueweather/repositories/CityRepositoryImpl.kt

```diff
@@ -0,0 +1,61 @@
+package com.rektstudios.trueweather.repositories
+
+import com.rektstudios.trueweather.data.local.CityItem
+import com.rektstudios.trueweather.data.local.IRealmDao
+import com.rektstudios.trueweather.data.mapper.toCityData
+import com.rektstudios.trueweather.data.remote.MapBoxApiService
+import com.rektstudios.trueweather.data.remote.reponses.mapbox.SearchResponse
+import com.rektstudios.trueweather.other.Constants.USER_UUID
+import com.rektstudios.trueweather.other.Resource
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class CityRepositoryImpl @Inject constructor(
+    private val mapBoxApiService: MapBoxApiService,
+    private val realmDao: IRealmDao
+): ICityRepository {
+    private suspend fun getSuggestionsFromApi(searchQuery: String, country: String = ""): Resource<SearchResponse> {
+        return try {
```

Comment:
please check how it works

Reply:
The catch block does trigger on network disconnection. As it it not possible to test for errors on the server side, I temporarily conclude that it is only for handling network errors.